app/src/main/java/com/summer/practice/tvtracker/DetailsActivity.kt

```diff
@@ -0,0 +1,32 @@
+package com.summer.practice.tvtracker
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import com.summer.practice.tvtracker.databinding.ActivityDetailsBinding
+
+class DetailsActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val binding = ActivityDetailsBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        binding.addToFavoritesButton.setOnClickListener {
+
+            //TODO add 'save as favorite' logic
+
+            val text = "Saved to favorites"
+            val duration = Toast.LENGTH_SHORT
+            val toast = Toast.makeText(applicationContext, text, duration)
+            toast.show()
+        }
+
+        binding.backArrow.setOnClickListener {
+            val intent = Intent(this, MainActivity::class.java)
+            startActivity(intent)
```

Comment:
this click action should not take you in this way to MainActivity.

The ideal backstack will look like this:
```SplashActivity -> LoginActivity -> MainActivity -> DetailsActivity```

In this case, when going back from DetailsActivity it will just close this activity and the MainActivity will be now in foreground. 

Using your approach by navigating to the MainActivity the backstack will look like this:
```SplashActivity -> LoginActivity -> MainActivity -> DetailsActivity -> MainActivity```

As you can see, the MainActivity is doubled and this is not a desired behavior. Please take a look into this [article](https://developer.android.com/guide/components/activities/tasks-and-back-stack) about back stack and let's discover what `onBackPressed()` method does.